Snakefile-sorting-hat

```diff
@@ -108,26 +108,32 @@ rule get_experiment_metadata:
         export ATLAS_PROD={params.atlas_prod}
         export ATLAS_EXPS={params.atlas_exps}
         source {workflow.basedir}/bin/metadata_retrieval_functions.sh
-        perl {workflow.basedir}/bin/get_exptype_contrasts_assays.pl {wildcards.accession} {input.config_file} {output.metadata}
-        if [ -e "{wildcards.accession}/{wildcards.accession}.condensed-sdrf.tsv" ]; then
-           echo "Trying with condensed..."
-           organism=$(get_organism_from_condensed_sdrf "{wildcards.accession}/{wildcards.accession}.condensed-sdrf.tsv")
-        elif [ -e "{wildcards.accession}/{wildcards.accession}-configuration.xml" ]; then
-           echo "Trying with perl (produces organism with capitals and spaces)"
-           organism=$(perl {workflow.basedir}/bin/get_experiment_info.pl --experiment {wildcards.accession} --xmlfile {wildcards.accession}/{wildcards.accession}-configuration.xml --organism | sed 's/ /_/g' | tr '[:upper:]' '[:lower:]')
+
+        gff_file=$( find {input.gtf_dir}/$organism -iname "$organism.*.gff3" | head -n1)
+        if [ -z "${gff_file}" ]; then
+            echo "gff not found. Yaml not created"  1>&2
+            exit 1
         else
-           echo "Nothing worked..."
-           >&2 echo "Can't retrieve organism: neither {wildcards.accession}/{wildcards.accession}.condensed-sdrf.tsv nor {wildcards.accession}/{wildcards.accession}-configuration.xml not found "
-           exit 1
+            perl {workflow.basedir}/bin/get_exptype_contrasts_assays.pl {wildcards.accession} {input.config_file} {output.metadata}
+            if [ -e "{wildcards.accession}/{wildcards.accession}.condensed-sdrf.tsv" ]; then
+                echo "Trying with condensed..."
+                organism=$(get_organism_from_condensed_sdrf "{wildcards.accession}/{wildcards.accession}.condensed-sdrf.tsv")
+            elif [ -e "{wildcards.accession}/{wildcards.accession}-configuration.xml" ]; then
+                echo "Trying with perl (produces organism with capitals and spaces)"
+                organism=$(perl {workflow.basedir}/bin/get_experiment_info.pl --experiment {wildcards.accession} --xmlfile {wildcards.accession}/{wildcards.accession}-configuration.xml --organism | sed 's/ /_/g' | tr '[:upper:]' '[:lower:]')
+            else
+                echo "Nothing worked..."
+                >&2 echo "Can't retrieve organism: neither {wildcards.accession}/{wildcards.accession}.condensed-sdrf.tsv nor {wildcards.accession}/{wildcards.accession}-configuration.xml not found "
+                exit 1
+            fi
+            magetabfiles=$( perl {workflow.basedir}/bin/get_experiment_info.pl --experiment {wildcards.accession} --xmlfile {wildcards.accession}/{wildcards.accession}-configuration.xml --magetabfiles)
+            idf=$( echo $magetabfiles | awk -F',' '{{ print $1 }}' )
+            sdrf=$( echo $magetabfiles | awk -F',' '{{ print $2 }}' )
+            echo "organism: $organism" >> {output.metadata}
```

Comment:
Maybe we should check in the same way and exit with 1 if organism, idf or sdrf files are not well defined, as that would mean a failure later on in the analysis parts.